_posts/2018-11-06-mongodb-europe-018.md

```diff
@@ -0,0 +1,408 @@
+> Data should be stored in the same way it is used -- MongoDB expert
+
+# [Streaming data pipelines with MongoDB and Kafka at AO](https://sched.co/FmAp){:target="_blank" rel="noopener noreferrer"}
+[A.O.](https://ao.com/){:target="_blank" rel="noopener noreferrer"}
+
+
+AO wanted to solve the issue of having data locked in different places so they wanted a [Single Customer View](https://en.wikipedia.org/wiki/Single_customer_view/){:target="_blank" rel="noopener noreferrer"}.
+The idea was to get the data from different places, like data stored in legacy databases or messages going through queues, and consolidate this in MongoDB.
+The data could be the usual customer data and phone calls with customer care, till the parcels moving through the warehouse and up until the doorstep.
+They wanted to get the data while it's hot, not in hourly or daily (or worse...) batches.
+They decided to use MongoDB to build up this materialised view of all different data streams, and Atlas to be able to focus on the application and not the database administration.
+
+The vast majority of the data resides in MsSql databases.
+Extraction happens with Kafka Connect SQL CDC to generate a stream of all create, update and delete operations into a stream and push it to Kafka.
+All with a simple piece of configuration like this : 
+
+```json
+{
+  "name" : "msSqlSourceConnector1",
+  "config" : {
+    "connector.class" : "io.confluent.connect.cdc.mssql.MsSqlSourceConnector",
+    "tasks.max" : "1",
+    "initial.database" : "testing",
+    "username" : "cdc",
+    "password" : "secret",
+    "server.name" : "db-01.example.com",
+    "server.port" : "1433"
+  }
+}
+```
+
+They use Avro for the schema definition in combination with a schema-registry.
+Interested clients can then read the data of the topics and do their single-view-thing on the data and save it to MongoDB.
+The view is being build up, message per message.
+Afterwards this view in  MongoDB is then pushed back to Kafka as another stream to provide this data to interested parties.
+This avoids locking the data in one place.
+
+To finish it of they shared some lessons learned :
+* Watch out for frameworks generating queries.  
+   They can create bad performing aggregations or queries.  
+   For them it was better to write some queries explicitly.  
+* Use custom _id for unique determination of your model, it saves an index and RAM
+* Watch out for unbounded document growth.
+
+
+# [MongoDB Enterprise Operator for Kubernetes at Amadeus](https://sched.co/FmAc){:target="_blank" rel="noopener noreferrer"}
+
+Amadeus is the world's largest technology company dedicated to the travel industry.
+They have developed an impressive MongoDB farm - a large environment with 100 clusters, some of which run more than 100 shards, some of which run 100TB MongoDB databases.
+Amadeus processes more than 1 trillion availability requests per day. 
+For each single search you do on a website they receive 500.000 availability requests.
```

Comment:
hier mis ik wat context.
Is dat het zoeken op een boekingssite of iets van een reisbureau?

Reply:
voldoende als ik gewoon 'flight' ertussen zet

> Amadeus processes more than 1 trillion flight availability requests per day.

Reply:
ja, dat maakt het idd mss wel wat duidelijker